lib/matplotlib/gridspec.py

```diff
@@ -467,19 +467,12 @@ def tight_layout(self, figure, renderer=None,
             coordinates that the whole subplots area (including labels) will
             fit into. Default (None) is the whole figure.
         """
-
-        subplotspec_list = _tight_layout.get_subplotspec_list(
-            figure.axes, grid_spec=self)
-        if None in subplotspec_list:
-            _api.warn_external("This figure includes Axes that are not "
-                               "compatible with tight_layout, so results "
-                               "might be incorrect.")
-
+        from ._tight_layout import get_subplotspec_list
```

Comment:
We already have `_tight_layout` imported, and this call would fit in a line...

Reply:
indeed.